### PR TITLE
New-app: more helpful message when not generating a service

### PR DESCRIPTION
**Pull request by csrwng**
Fixes https://github.com/openshift/origin/issues/2707

### CODE REVIEW

**Comment by csrwng** [comment]
@sosiouxme ptal

**Comment by sosiouxme** [comment]
Thanks, that looks very helpful.
Thinks it's safe to [merge]

File: pkg/generate/app/pipeline.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func AddServices(objects Objects) Objects {
 				}
 			}
 			if len(svc.Spec.Ports) == 0 {
-				glog.Warningf("DeploymentConfig %q: Cannot create a service with no ports", t.Name)
+				glog.Warningf("A service will not be generated for DeploymentConfig %q because no exposed ports were detected. Use 'osc expose dc %q --port=[port] --generator=service/v1' to create a service.", t.Name, t.Name)
```
Comment (kargakis) [inline pkg/generate/app/pipeline.go]: +1 for this message, the name of the dc in the command example shouldn't be quoted though...

Comment (csrwng) [inline pkg/generate/app/pipeline.go]: it works with the quoted name, and it helps the shell interpret it as a single argument

Comment (kargakis) [inline pkg/generate/app/pipeline.go]: k, nice to know

Comment (liggitt) [inline pkg/generate/app/pipeline.go]: What a name with spaces ever be accepted by the API? If not, the quotes seem unnecessary

Comment (csrwng) [inline pkg/generate/app/pipeline.go]: hmm, true

Comment (csrwng) [inline pkg/generate/app/pipeline.go]: @kargakis @liggitt - I'll add a follow up pr to remove the quotes... I tried to think of an instance where we'd want them and I can't; the command is easier to read without them.

**Pull request merged**